src/vec.rs

```diff
@@ -568,6 +569,28 @@ where
         other.into_iter().for_each(|i| _ = self.append(i))
     }
 
+    /// Try Append a new member to the end of the `identified_vec`, if the `identified_vec` already contains the element a InsertionError will be returned.
+    ///
+    /// - Parameter item: The element to add to the `identified_vec`.
+    /// - Returns: Either a Ok() with a pair `(inserted, index)`, where `inserted` is a Boolean value indicating whether
+    ///   the operation added a new element, and `index` is the index of `item` in the resulting
+    ///   `identified_vec`. If the given ID or Value pre-exists within the collection the function call returns a InsertionFailure Error specifying the reason.
+    /// - Complexity: The operation is expected to perform O(1) copy, hash, and compare operations on
+    ///   the `ID` type, if it implements high-quality hashing.
+    #[inline]
+    pub fn try_append(&mut self, element: Element) -> Result<(bool, usize), InsertionFailure> {
+        let id = self.id(&element);
+        if self.index_of_id(&id).is_some() {
+            return Err(InsertionFailure::ElementWithSameIDFound);
+        }
+
+        if self.contains(&element) {
+            return Err(InsertionFailure::ElementWithSameValueFound);
```

Comment:
`contains` is not what you want, it only checks if a value for the ID exists, which is the check you did above already.

Same Element can only be asserted by checking the value, which you cant in this scope, since `Element` does not impl `Eq`.

So actually, you want / need TWO different methods, one where `Element: Eq` and one without. So probably something like (and with docs...)

```rust

impl<I, E> IdentifiedVec<I, E>
where
    E: Eq,
    I: Eq + Hash + Clone + Debug,
{
    #[inline]
    pub fn try_append_unique_element(&mut self, element: E) -> Result<(bool, usize), InsertionFailure> {
        let id = self.id(&element);

        if let Some(value) = self.get(&id) {
            if value == &element {
                return Err(InsertionFailure::ElementWithSameValueFound);
            } else {
                return Err(InsertionFailure::ElementWithSameIDFound);
            }
        };

        Ok(self.append(element))
    }
}
impl<I, E> IdentifiedVec<I, E>
where
    I: Eq + Hash + Clone + Debug,
{
    #[inline]
    pub fn try_append(&mut self, element: E) -> Result<(bool, usize), InsertionFailure> {
        let id = self.id(&element);

        if self.contains_id(&id) {
             return Err(InsertionFailure::ElementWithSameIDFound);
        }

        Ok(self.append(element))
    }
}
```

Reply:
Duly noted and fair points. I've re-created all functions with their corresponding tests and documentation to match this.